src/condor_dagman/node.h

```diff
@@ -36,643 +34,467 @@
-//
 #include "debug.h"
 #include "script.h"
 
-
-typedef int NodeID_t;
-#define NO_ID -1
-
-typedef int EdgeID_t;
-#define NO_EDGE_ID -1
-
+// Node Type
 enum NodeType {
 	JOB,
 	FINAL,
 	PROVISIONER,
 	SERVICE
 };
 
-#define EXEC_MASK 0x1
-#define ABORT_TERM_MASK 0x2
-#define IDLE_MASK 0x4 // set when proc is idle, formerly a separate _isIdle vector
-#define HOLD_MASK 0x8 // set when proc is held, formerly a separate _onHold vector
-
-/**  The job class represents a job in the DAG and its state in the HTCondor
-     system.  A job is given a name, a CondorID, and three queues.  The
-     parents queue is a list of parent jobs that this one depends on.  That
-     queue never changes once set.  The waiting queue is the same as the
-     parents queue, but shrinks to emptiness has the parent jobs complete.
-     The children queue is a list of child jobs that depend on this one.
-     Once set, this queue never changes.<p>
-
-     From DAGMan's point of view, a job has six basic states in the
-     HTCondor system (refer to the Node::status_t enumeration).  It
-     starts out as READY, meaning that it has not yet been submitted.
-     If the job has a PRE script defined it changes to PRERUN while
-     that script executes.  The job's state changes to SUBMITTED once
-     in the HTCondor system.  If it completes successfully and the job
-     has a POST script defined it changes to POSTRUN while that script
-     executes, otherwise if it completes successfully it is DONE, or
-     is in the ERROR state if it completes abnormally.  Note that final
-     nodes are a special case -- they are created with a state of
-     NOT_READY<p>
-
-     The DAG class will control the job by modifying and viewing its
-     three queues.  Once the WAITING queue becomes empty, the job
-     state either changes to PRERUN while the job's PRE script runs,
-     or the job is submitted and its state changes immediately to
-     SUBMITTED.  If the job completes successfully, its state is
-     changed to DONE, and the children (listed in this jobs CHILDREN
-     queue) are put on the DAG's ready list.  */
-class Node {
-  public:
-  
-	// true when node has no parents at this time
-	bool NoParents() const {
-		// Once we are done with AdjustEdges this should agree
-		// but some code checks for parents during parse time (see the splice code)
-		// so we check _numparents (set at parse time) and also _parent (set by AdjustEdges)
-		return _parent == NO_ID && _numparents == 0;
-	}
-
-	// true when node has no children at this time
-	bool NoChildren() const {
-		return _child == NO_ID;
-	}
+typedef int NodeID_t;
+typedef int EdgeID_t;
 
-	// returns the number of children.  NOTE: this is not guaranteed to be fast!
-	int CountChildren() const;
+const NodeID_t NO_ID = -1;
+const EdgeID_t NO_EDGE_ID = -1;
+
+// Job proc event masks
+const int EXEC_MASK = (1 << 0);
+const int ABORT_TERM_MASK = (1 << 1);
+const int IDLE_MASK = (1 << 2);
+const int HOLD_MASK = (1 << 3);
```

Comment:
I like the use of `const int` here, but should we keep the comments from the previous code?